src/backend/dist.js

```diff
@@ -223,62 +223,90 @@ module.exports = {
     unzipper.on('extract', function(log) {
       logger.addLog('Info', 'zip successfully extracted', socket);
 
-      fs.readdir(appPath + '/zip' , function(err, list){
-        if(err) {
-          logger.addLog('Error', 'Error while reading directory', socket, err);
-          return done(err);
+        async.series([
+
+          function(callback) {
+            fs.readdir( appPath + '/zip/images/sponsors', function(err, list){
+              if(err) {
+                logger.addLog('Error', 'Error while reading sponsors directory', socket, err);
```

Comment:
Please note that it might be possible that there is no sponsors or speakers folder inside the image directory. Ehealth Africa zip is an example of this. So we shouldn't stop the generation process and display error if we found them missing. We can add an 'Info' Statement saying that 'No sponsors/speakers folder found' and move to the next step.